packages/headless/src/controllers/commerce/context/cart/headless-cart.test.ts

```diff
@@ -1,87 +1,140 @@
+import {setItems} from '../../../../features/commerce/context/cart/cart-actions';
 import {cartReducer} from '../../../../features/commerce/context/cart/cart-slice';
-import {MockCommerceEngine} from '../../../../test/mock-engine';
-import {buildMockCommerceEngine} from '../../../../test/mock-engine';
-import {buildCart, Cart} from './headless-cart';
+import {contextReducer} from '../../../../features/commerce/context/context-slice';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  MockedCommerceEngine,
+  buildMockCommerceEngine,
+} from '../../../../test/mock-engine-v2';
+import {buildCart, Cart, CartOptions} from './headless-cart';
+import {
+  itemsSelector,
+  totalPriceSelector,
+  totalQuantitySelector,
+} from './headless-cart-selectors';
 
-describe('headless commerce cart', () => {
-  const options = {
-    items: [
-      {
-        productId: 'initial-product-id',
-        quantity: 2,
-      },
-    ],
-  };
-
-  const item = {
-    productId: 'new-product-id',
-    quantity: 1,
-  };
+jest.mock('../../../../features/commerce/context/cart/cart-actions');
+jest.mock('./headless-cart-selectors');
 
+describe('headless commerce cart', () => {
+  let engine: MockedCommerceEngine;
+  let options: CartOptions;
   let cart: Cart;
-  let engine: MockCommerceEngine;
 
-  beforeEach(() => {
-    engine = buildMockCommerceEngine();
+  function initEngine(preloadedState = buildMockCommerceState()) {
+    engine = buildMockCommerceEngine(preloadedState);
+  }
+
+  function initCart() {
     cart = buildCart(engine, {options});
+  }
+
+  beforeEach(() => {
+    initEngine();
+    jest.resetAllMocks();
+    options = {
+      items: [
+        {
+          productId: 'product-id-1',
+          quantity: 2,
+          name: 'product-name-1',
+          price: 100,
+        },
+        {
+          productId: 'product-id-2',
+          quantity: 4,
+          name: 'product-name-2',
+          price: 25,
+        },
+      ],
+    };
+    initCart();
   });
 
-  const expectContainAction = (action: Action) => {
-    const found = engine.actions.find((a) => a.type === action.type);
-    expect(engine.actions).toContainEqual(found);
-  };
-
-  const expectContainActionWithPayload = (
-    action: Action,
-    payload: string | object
-  ) => {
-    expect(engine.actions).toEqual(
-      expect.arrayContaining([
-        expect.objectContaining({
-          type: action.type,
-          payload:
-            typeof payload === 'string'
-              ? expect.stringContaining(payload)
-              : expect.objectContaining(payload),
-        }),
-      ])
-    );
-  };
+  it('initializes', () => {
+    expect(cart).toBeTruthy();
+  });
+
+  it('exposes a #subscribe method', () => {
+    expect(cart.subscribe).toBeTruthy();
+  });
 
   it('adds the correct reducers to engine', () => {
     expect(engine.addReducers).toHaveBeenCalledWith({
       cart: cartReducer,
+      context: contextReducer,
```

Comment:
Why do we load the context reducer?

If someone uses a cart controller, wouldn't they have already loaded a context controller?

Reply:
Yep you're right, I don't think we need to load this reducer after all